app/src/components/notification/Notification.tsx

```diff
@@ -1,28 +1,127 @@
-import React from "react";
-import { useDispatch } from "react-redux";
+import React, { useEffect, useRef, useState } from "react";
+import { shallowEqual, useDispatch, useSelector } from "react-redux";
 import { Link } from "react-router-dom";
-import { markNotificationAsReadAsync } from "../../pages/notifications/notificationSlice";
+import {
+  dismissNotificationAsync,
+  markNotificationAsReadAsync,
+  selectOpenMenus,
+  setCloseContextMenu,
+  setContextMenuOpen,
+} from "../../pages/notifications/notificationSlice";
 
 import Avatar from "../avatar";
 import ContextMenuButton from "./context-menu-button";
 import "./notification-component-style.css";
 import { INotificationCardData } from "./notification.types";
+import { useSwipeable } from "react-swipeable";
+import NotificationContextMenu from "./context-menu";
+import { NotificationContextMenuAction } from "./context-menu/Notification-context-menu";
 
 function NotificationElement(props: INotificationCardData) {
   const dispatch = useDispatch();
-
+  const [isSwiping, setIsSwiping] = useState<boolean>(false);
+  const [swipingObject, setSwipingObject] = useState<any>();
+  const [contextMenuVisible, setContextMenuVisible] = useState<boolean>(false);
   const handleOnLinkClick = (notificationId: string) => {
-    // Here we need to mark the notification as read
     dispatch(markNotificationAsReadAsync(notificationId));
   };
+  const openMenus = useSelector(selectOpenMenus, shallowEqual);
+
+  useEffect(() => {
+    if (swipingObject) {
+      swipingObject.animate(
+        [{ transform: "translateX(0px)" }, { transform: "translateX(-500px)" }],
+        { duration: 400 }
+      );
+      swipingObject.style["margin-left"] = `-1300px`;
+      if (props.id) {
+        dispatch(dismissNotificationAsync(props.id));
+        setIsSwiping(false);
+      }
+    }
+  }, [dispatch, swipingObject, props.id]);
+
+  const swipeHandlers = useSwipeable({
+    preventDefaultTouchmoveEvent: true,
+    trackMouse: true,
+    onSwipedLeft: (data: any) => {
+      if (!isSwiping && data.velocity < 1) {
+        data.event.currentTarget.style["margin-left"] = `0px`;
```

Comment:
`data.event.currentTarget` works on touch events but is null when triggered by mouse pointer

Reply:
Just looking for a bit of context for this one: Is this crashing for you? I was thinking that in mouse pointer contexts, the user would use the context menu.

Reply:
Ahh ok I see what is happening now!

Reply:
I did a null check. Just to confirm - do we want mouse-swiping in this scenario?

Reply:
Makes sense to not have mouse swiping here